Exiled.API/Features/GlobalPatchProcessor.cs

```diff
@@ -0,0 +1,161 @@
+
+                    if (patchGroup is null)
+                    {
+                        processor.Patch();
+                        continue;
+                    }
+
+                    if (string.IsNullOrEmpty(patchGroup.GroupId))
+                        throw new ArgumentNullException("GroupId");
+
+                    if (string.IsNullOrEmpty(groupId) || patchGroup.GroupId != groupId)
+                        continue;
+
+                    isPatchGroup = true;
+                    processor.Patch();
+                }
+
+                if (!isPatchGroup)
+                    return harmony;
+
+                foreach (MethodBase methodBase in harmony.GetPatchedMethods())
+                {
+                    if (PatchedGroupMethods.TryGetValue(methodBase, out HashSet<string> ids))
+                        ids.Add(groupId);
+                    else
+                        PatchedGroupMethodsValue.Add(methodBase, new() { groupId });
+
+#if DEBUG
+                    Log.Debug($"Target method ({methodBase.Name}) has been successfully patched.");
+#endif
+                }
+
+#if DEBUG
+                MethodBase callee = new StackTrace().GetFrame(1).GetMethod();
+                Log.Debug($"Patching completed. Requested by: ({callee.DeclaringType.Name}::{callee.Name})");
+#endif
+                return harmony;
+            }
+            catch (Exception ex)
+            {
+                MethodBase callee = new StackTrace().GetFrame(1).GetMethod();
+                Log.Error($"Callee ({callee.DeclaringType.Name}::{callee.Name}) Patching failed!, " + ex);
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Unpatches methods by patching them with zero patches.
+        /// </summary>
+        /// <param name="id">The Harmony instance id.</param>
+        /// <param name="groupId">The target group to include.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <see cref="PatchGroupAttribute.GroupId"/> is <see langword="null"/> or empty.</exception>
+        public static void UnpatchAll(string id = "", string groupId = null)
+        {
+            Harmony harmony = new(id);
+            foreach (MethodBase methodBase in Harmony.GetAllPatchedMethods().ToList())
+            {
+                PatchProcessor processor = harmony.CreateProcessor(methodBase);
+
+                Patches patchInfo = Harmony.GetPatchInfo(methodBase);
+                if (!patchInfo.Owners.Contains(id))
+                    continue;
+
+                PatchGroupAttribute patchGroup = methodBase.GetCustomAttribute<PatchGroupAttribute>();
+                if (patchGroup is null)
+                    goto Unpatch;
+
+                if (string.IsNullOrEmpty(patchGroup.GroupId))
+                    throw new ArgumentNullException("GroupId");
+
+                if (string.IsNullOrEmpty(groupId) || patchGroup.GroupId != groupId)
+                    continue;
+
+                Unpatch:
```

Comment:
Why not just make a private unpatch method to prevent goto shenanigans

Reply:
No